### DISCUSSION TITLE
Require authorization to run introspection

### DISCUSSION THREAD

**Discussion by mfulton26** [MichalLytek/type-graphql#1308] [Q&A]
Is there a way to enable introspection but only honor such requests if the authenticated user has a certain role (i.e. is authorized)?

**Comment by karladler** [comment]
depends on your architecture, but with this function you might be able to detect an introspection query and then you can add it to any middleware layer on your backend which checks you auth token or whatever...

```typescript
import { parse, print, getIntrospectionQuery } from 'graphql';

const introspectionQuery = print(parse(getIntrospectionQuery()));
const isIntrospectionQuery = (context: Context) => context.request.body?.query === introspectionQuery;
```

**Reply by mfulton26** [reply]
I like the thinking here; I'd prefer more to have something built into `type-graphql`'s authorization features but it might not be too difficult to implement in my own project

thanks

**Reply by mfulton26** [reply]
https://github.com/MichalLytek/type-graphql/issues/1341

**Comment by savaparoski** [comment]
I have the same issue, @mfulton26 have you found a solution?

Everyone suggests some workarounds on this issue, is there a proper solution?
